### PR TITLE
crystal: init at 1.0.0

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Crystal hit version 1!!

Update a few crystal things that otherwise break as a consequence.

We drop the old versions 0.31 and 0.32 from the top-level for now, but let's just leave them available for the time being. Before 21.05 is branched, this needs to be cleaned up and hopefully by then, most things will be updated to work with crystal 1.0

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
- [ ] 
Cc: @NixOS/crystal-lang

### CODE REVIEW

**Comment by peterhoeg** [comment]
`scry` is left out for now due to https://github.com/NixOS/nixpkgs/issues/118017

**Comment by peterhoeg** [comment]
scry is now included.

File: pkgs/development/compilers/mint/default.nix [hunk 1]
```diff
@@ -1,15 +1,18 @@
-{ lib, fetchFromGitHub, crystal_0_33, openssl }:
+{ lib, fetchFromGitHub, crystal_0_36, openssl }:
 
-let crystal = crystal_0_33;
-in crystal.buildCrystalPackage rec {
-  version = "0.9.0";
+let
+  crystal = crystal_0_36;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/mint/default.nix:4]: crystal should be overwritten in top-level.

**Comment by peterhoeg** [comment]
Generally yes, but this crystal situation is a little different.

Crystal just hit 1.0 and several of the crystal packages we have depend on various older versions. It's *much* more obvious to someone looking at this what is going on and it will only be for a while until everything works properly with 1.0 in which case any specific overrides for older/legacy packages will be done as they should in `all-packages.nix`

**Pull request merged**